metrics.go

```diff
@@ -33,63 +33,61 @@ type Metrics struct {
 
 	// Number of spans successfully reported
-	ReporterSuccess metrics.Counter `metric:"reporter-spans" tags:"state=success"`
+	ReporterSuccess metrics.Counter `metric:"reporter_spans_reported"`
 
-	// Number of spans in failed attempts to report
-	ReporterFailure metrics.Counter `metric:"reporter-spans" tags:"state=failure"`
+	// Number of spans not reported due to some failure
+	ReporterFailure metrics.Counter `metric:"reporter_spans_failed"`
 
 	// Number of spans dropped due to internal queue overflow
-	ReporterDropped metrics.Counter `metric:"reporter-spans" tags:"state=dropped"`
+	ReporterDropped metrics.Counter `metric:"reporter_spans_dropped"`
 
 	// Current number of spans in the reporter queue
-	ReporterQueueLength metrics.Gauge `metric:"reporter-queue"`
+	ReporterQueueLength metrics.Gauge `metric:"reporter_queue_length"`
 
 	// Number of times the Sampler succeeded to retrieve sampling strategy
-	SamplerRetrieved metrics.Counter `metric:"sampler" tags:"state=retrieved"`
+	SamplerRetrieved metrics.Counter `metric:"sampler_retrieved"`
 
 	// Number of times the Sampler succeeded to retrieve and update sampling strategy
-	SamplerUpdated metrics.Counter `metric:"sampler" tags:"state=updated"`
+	SamplerUpdated metrics.Counter `metric:"sampler_updated"`
 
 	// Number of times the Sampler failed to update sampling strategy
-	SamplerUpdateFailure metrics.Counter `metric:"sampler" tags:"state=failure,phase=updating"`
+	SamplerUpdateFailure metrics.Counter `metric:"sampler_update_failures"`
 
 	// Number of times the Sampler failed to retrieve sampling strategy
-	SamplerQueryFailure metrics.Counter `metric:"sampler" tags:"state=failure,phase=query"`
+	SamplerQueryFailure metrics.Counter `metric:"sampler_query_failures"`
 
 	// Number of times baggage was successfully written or updated on spans.
-	BaggageUpdateSuccess metrics.Counter `metric:"baggage-update" tags:"result=ok"`
+	BaggageUpdateSuccess metrics.Counter `metric:"baggage_updates" tags:"result=ok"`
 
 	// Number of times baggage failed to write or update on spans.
-	BaggageUpdateFailure metrics.Counter `metric:"baggage-update" tags:"result=err"`
+	BaggageUpdateFailure metrics.Counter `metric:"baggage_updates" tags:"result=err"`
 
 	// Number of times baggage was truncated as per baggage restrictions.
-	BaggageTruncate metrics.Counter `metric:"baggage-truncate"`
+	BaggageTruncate metrics.Counter `metric:"baggage_truncated"`
 
 	// Number of times baggage restrictions were successfully updated.
-	BaggageRestrictionsUpdateSuccess metrics.Counter `metric:"baggage-restrictions-update" tags:"result=ok"`
+	BaggageRestrictionsUpdateSuccess metrics.Counter `metric:"baggage_restrictions_updates" tags:"result=ok"`
 
 	// Number of times baggage restrictions failed to update.
-	BaggageRestrictionsUpdateFailure metrics.Counter `metric:"baggage-restrictions-update" tags:"result=err"`
+	BaggageRestrictionsUpdateFailure metrics.Counter `metric:"baggage_restrictions_updates" tags:"result=err"`
 }
```

Comment:
@pavolloffay @objectiser penny for your thoughts.

Reply:
+1 - I assume the names will be consistent across all clients?

Reply:
As much as possible. I don't think we have tag-based APIs in Python and Node.js clients.

Another option would be to drop tags completely and just encode them in the name, e.g. `traces_started_sampled`. It makes it harder to aggregate if say you don't care about the "is_sampled" dimension.

Reply:
Better to have tags - is there any reason they can't be added to python and node clients?